presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -47,34 +46,39 @@
  * requests. This port number has to be the same as to which your externally launched process listens.
  * Example: -DNATIVE_PORT=7777.
  * When NATIVE_PORT is specified, PRESTO_SERVER argument is not requires and is ignored if specified.
+ * <p>
+ * For test queries requiring shuffle, the disk-based local shuffle will be used. For more details of the local shuffle,
+ * Please refer to:
+ * <a href="https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/operators/LocalPersistentShuffle.h">Local Shuffle</a>
  */
 public class TestPrestoSparkNativeExecution
         extends AbstractTestPrestoSparkQueries
 {
     @Test
-    public void testMapOnlyQueries()
+    public void testQueries()
     {
+        // Single stage queries
         assertQuerySucceeds("SELECT * FROM orders");
         assertQuery("SELECT orderkey, custkey FROM orders WHERE orderkey <= 200");
         assertQuery("SELECT nullif(orderkey, custkey) FROM orders");
 
+        // Aggregation queries
+        assertQuery("SELECT orderkey, custkey FROM orders ORDER BY orderkey LIMIT 4");
+        assertQuery("SELECT count(*) c FROM lineitem WHERE partkey % 10 = 1 GROUP BY partkey");
+
+        // Join queries
+        assertQuery("SELECT count(*) FROM orders o, lineitem l WHERE o.orderkey = l.orderkey AND o.orderkey % 2 = 1");
+
+        // Failure cases
         assertQueryFails("SELECT sequence(1, orderkey) FROM orders",
                 ".*Scalar function name not registered: presto.default.sequence.*");
         assertQueryFails("SELECT orderkey / 0 FROM orders", ".*division by zero.*");
-        assertQueryFails("SELECT orderkey, custkey FROM orders LIMIT 4", ".*Failure in LocalWriteFile: path .* already exists.*");
     }
 
-    // TODO: re-enable the test once the shuffle integration is ready.
-    @Ignore
-    @Test(priority = 2, dependsOnMethods = "testMapOnlyQueries")
+    @Test
     public void testNativeExecutionShuffleManager()
```

Comment:
Do we need this test? Let's remove it.

Reply:
We use this test to test the functionality of the `PrestoSparkNativeExecutionShuffleManager`

Reply:
What do we test there? Should this be a separate test?

Reply:
Let me move it to a separated test suite. It's testing the `PrestoSparkNativeExecutionShuffleManager` can correctly bookkeep all required shuffle meta information.